index.js

```diff
@@ -53,7 +53,7 @@ module.exports = {
         self.i18nStaticIds = self.i18nStaticIds || {};
 
         if (self.i18nStaticIds[aposLocale] !== req.data.global.i18nStaticId) {
-          const namespaces = await self.findPiecesAndGroupByNamespace(aposLocale);
+          const namespaces = await self.apos.cache.get(req.locale, 'i18n-static') || await self.findPiecesAndGroupByNamespace(aposLocale);
```

Comment:
Please make sure the precedence of `await` is correct in this situation. If you're not sure use parentheses. I know the precedence of `await` trips me up a lot personally but you may be correct here.